msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java

```diff
@@ -1898,16 +1898,20 @@ public String processPvtMsgSaveDraft() {
 
     PrivateMessage dMsg = null;
     if(getDetailMsg() != null && getDetailMsg().getMsg() != null && getDetailMsg().getMsg().getDraft()){
-    	dMsg =constructMessage(true, getDetailMsg().getMsg()) ;
+	    dMsg =constructMessage(true, getDetailMsg().getMsg()) ;
+	    PrivateMessageSchedulerService.removeScheduledReminder(dMsg.getId());
     }else{
-    	dMsg =constructMessage(true, null) ;
+	    dMsg =constructMessage(true, null) ;
     }
     dMsg.setDraft(Boolean.TRUE);
     dMsg.setDeleted(Boolean.FALSE);
     dMsg.setExternalEmail(booleanEmailOut);
     dMsg.setScheduler(booleanSchedulerSend);
-    if(booleanSchedulerSend)
+    if(booleanSchedulerSend){
```

Comment:
There's a small spacing issue, but you can also use this:

dMsg.setScheduledDate(booleanSchedulerSend ? openDate : null);

Reply:
Done!